File: 0905-sort-array-by-parity/0905-sort-array-by-parity.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int[] res = new int[nums.length];
        int l = 0, r = res.length-1;
        for(int idx=0;idx<nums.length;idx++){
            if(nums[idx]%2==0) {
                res[l] = nums[idx];
                l++;
            } else {
                res[r] = nums[idx];
                r--;
            }
        }
        return res;
    }
}

File: 0905-sort-array-by-parity/0905-sort-array-by-parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        if len(nums) == 1:
            return nums
        else:
            output = [0]*len(nums)
            l = 0
            r = len(output)-1
            for idx in range(len(nums)):
                if nums[idx]%2 == 0:
                    output[l]=nums[idx]
                    l+=1
                else:
                    output[r]= nums[idx]
                    r-=1
        return output       